pkg/scale/decode_test.go

```diff
@@ -132,6 +132,40 @@ func Test_decodeState_decodeSlice(t *testing.T) {
 	}
 }
 
+func Test_decodeState_decodeMap(t *testing.T) {
+	mapTests = tests{
+		{
+			name: "testMap1",
+			in:   map[int8][]byte{2: []byte("some string")},
+			want: []byte{4, 2, 44, 115, 111, 109, 101, 32, 115, 116, 114, 105, 110, 103},
+		},
+		{
+			name: "testMap2",
+			in: map[int8][]byte{
+				2:  []byte("some string"),
+				16: []byte("lorem ipsum"),
+			},
+			want: []byte{
+				8, 2, 44, 115, 111, 109, 101, 32, 115, 116, 114, 105, 110, 103, 16, 44, 108, 111, 114, 101, 109, 32,
+				105, 112, 115, 117, 109,
+			},
+		},
+	}
+
+	for _, tt := range mapTests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
you should pin the tt variable to the loop context otherwise it can cause inconsistency
```suggestion
	for _, tt := range mapTests {
	        tt := tt
		t.Run(tt.name, func(t *testing.T) {
```

Reply:
Only if `t.Parallel()` is used on the subtest :wink: